src/Tasks/Exec.cs

```diff
@@ -611,18 +624,38 @@ protected internal override void AddCommandLineCommands(CommandLineBuilderExtens
 
                     // If for some crazy reason the path has a & character and a space in it
                     // then get the short path of the temp path, which should not have spaces in it
-                    // and then escape the &
                     if (batchFileForCommandLine.Contains("&") && !batchFileForCommandLine.Contains("^&"))
                     {
                         batchFileForCommandLine = NativeMethodsShared.GetShortFilePath(batchFileForCommandLine);
-                        batchFileForCommandLine = batchFileForCommandLine.Replace("&", "^&");
                     }
+
+                    StringBuilder fileName = StringBuilderCache.Acquire(batchFileForCommandLine.Length);
+
+                    // Escape any characters specified by the CharactersToEscape metadata, or '&'
+                    for (int i = 0; i < batchFileForCommandLine.Length; i++)
+                    {
+                        char c = batchFileForCommandLine[i];
+
+                        if (ShouldEscapeCharacter(c) && (i == 0 || batchFileForCommandLine[i - 1] != '^'))
```

Comment:
You said you wanted to put off fixing edge cases like not escaping a character preceded by an escaped ^, correct?

Reply:
Correct. at the moment we've only gotten a request for the parentheses case (user's username was something(dev) which seems reasonable). So this fix will account for _most_ other special characters, and I'll make an issue tracking that we don't account for '^''s that are already escaped and the other special cases that arise with it. If it gains traction we can worry about it then. For now this is a "better than it was before" PR.